### ISSUE TITLE
mod-inverse returns incorrect answer

### ISSUE THREAD

**Issue by ktakashi** [ktakashi/sagittarius-scheme#213]
**[Original report](https://bitbucket.org/ktakashi/sagittarius-scheme/issues/208) by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

This should return `84793287459004005994083570264676611930995373170935977255695558296701128546491` but `-1868565590369779796042771496011207097362409954010577056616565239046619145529369611450115909`.
```
#!scheme
(mod-inverse
  59791678501913488631701617161572303141620876383029885416585973023996318696896
  115792089210356248762697446949407573530086143415290314195533631308867097853951)
```
NB: this only happens 32 bit environment. More precisely 32bit long environment.

**Comment by ktakashi** [comment]
**Original comment by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

Fixed value

**Comment by ktakashi** [comment]
**Original comment by Takashi Kato (Bitbucket: [ktakashi](https://bitbucket.org/ktakashi), GitHub: [ktakashi](https://github.com/ktakashi)).**

----------------------------------------

Clearing stack area. Fixes #208)
